src/libcommonserver/utility/utility_win.cpp

```diff
@@ -112,15 +112,50 @@ static bool initTrusteeWithUserSID() {
 }
 
 static bool init_private() {
+    const std::string useGetRightsFallbackMethod = CommonUtility::envVarValue("KDRIVE_USE_GETRIGHTS_FALLBACK_METHOD");
+    if (!useGetRightsFallbackMethod.empty()) {
+        LOG_DEBUG(Log::instance()->getLogger(), "KDRIVE_USE_GETRIGHTS_FALLBACK_METHOD env is set, using fallback method");
+        IoHelper::_getAndSetRightsMethod = 1;
+        return true;
+    }
+
     if (Utility::_psid != nullptr) {
         return false;
     }
-
     initTrusteeWithUserSID();
 
-    const std::string useGetRightsFallbackMethod = CommonUtility::envVarValue("KDRIVE_USE_GETRIGHTS_FALLBACK_METHOD");
-    if (!useGetRightsFallbackMethod.empty()) {
-        LOG_DEBUG(Log::instance()->getLogger(), "Use getRights fallback method");
+    // Check getRights method performance
+    SyncPath tmpDir;
+    IoError ioError = IoErrorSuccess;
+    if (!IoHelper::tempDirectoryPath(tmpDir, ioError)) {
+        LOGW_WARN(Log::instance()->getLogger(),
+                  "Error in IoHelper::tempDirectoryPath: " << Utility::formatIoError(tmpDir, ioError));
+        return false;
+    }
+
+    bool read = true;
+    bool write = true;
+    bool execute = true;
+
+    auto start = std::chrono::high_resolution_clock::now();
+    for (int i = 0; i < 10; i++) {
+        if (!IoHelper::getRights(tmpDir, read, write, execute, ioError)) {
+            LOGW_WARN(Log::instance()->getLogger(), "Error in IoHelper::getRights: " << Utility::formatIoError(tmpDir, ioError));
+            return false;
+        }
+    }
+    auto end = std::chrono::high_resolution_clock::now();
+    auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
+
+    /* Averrage times for 10 calls to getRights with windows API:
```

Comment:
```suggestion
    /* Average time spent in getRights with windows API:
```
and then divide the subsequent time estimates by 10.

Reply:
Done 👍🏾 ([43d8601](https://github.com/Infomaniak/desktop-kDrive/pull/155/commits/43d8601f32d5a94a22f1da7e4fcab280de83f82a))